utils/parenthesesutils.go

```diff
@@ -36,7 +36,7 @@ func (p *ParenthesesSlice) IsPresent(index int) bool {
 }
 
 // Return true if at least one of the {i} in 'target' has corresponding parentheses in 'pattern'.
-func PlaceholdersUserd(pattern, target string) bool {
+func ArePlaceholdersUsed(pattern, target string) bool {
```

Comment:
I suggest reverting the name change. I'm not the "Are" prefix is actually a convention.